### DISCUSSION TITLE
How to get body for erroneously reported HTTP errors

### DISCUSSION THREAD

**Discussion by forthrin** [HoneyryderChuck/httpx#55] [Q&A]
https://www.moviesubtitles.org/search.php?q=the returns HTTP 500 (even in a browser), but -does- return a valid body.

How does one get this body, when the returned object `HTTPX::ErrorResponse` does not contain the `.body`property?

**Comment by HoneyryderChuck** [comment]
hi, what's the exact code snippet we're talking about? FWIW 500s will raise a [`HTTPX::HTTPError` on `.raise_for_status`, which contains the response object](https://rubydoc.info/gems/httpx/HTTPX/HTTPError)

**Comment by forthrin** [comment]
Ah, so a quick hack would be:

```
response = HTTPX.get('https://www.moviesubtitles.org/search.php?q=the')
response = response.response if response.error
puts response.body
```

It seems there is some sort of SSL error on the server: `#<OpenSSL::SSL::SSLError: SSL_read: unexpected eof while reading>`

**Comment by HoneyryderChuck** [comment]
`HTTPX::ErrorResponse` is not a `HTTPX::HTTPError`, and that type of ssl error will generate the former. Perhaps you want to reas [this](https://gitlab.com/os85/httpx/-/wikis/Error-Handling#error-pattern-matching)
